cirq/neutral_atoms/atom_devices.py

```diff
@@ -0,0 +1,423 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Iterable
+from numpy import sqrt
+from cirq import devices, ops, circuits, value
+from cirq.devices.grid_qubit import GridQubit
+from cirq.ops import MeasurementGate
+from cirq.value import Duration
+from cirq.neutral_atoms import convert_to_atom_gates
+
+
+@value.value_equality
+class AtomDevice(devices.Device):
+    """
+    A device with qubits placed on a grid.
+    """
+
+    def __init__(self,
+                 measurement_duration: Duration,
+                 gate_duration: Duration,
+                 control_radius: float,
+                 max_parallel_z: int,
+                 max_parallel_xy: int,
+                 max_parallel_c: int,
+                 qubits: Iterable[GridQubit])->None:
+        """
+        Initializes the description of the AQuA device.
+
+        Args:
+            measurement_duration: the maximum duration of a measurement.
+            gate_duration: the maximum duration of a gate
+            control_radius: the maximum distance between qubits for a controlled
+            gate. Distance is measured in units of the indices passed into the
+            GridQubit constructor.
+            max_parallel_z: The maximum number of qubits that can be acted on
+            in parallel by a Z gate
+            max_parallel_xy: The maximum number of qubits that can be acted on
+            in parallel by a local XY gate
+            max_parallel_c: the maximum number of qubits that can be acted on in
+            parallel by a controlled gate. Must be less than or equal to the
+            lesser of max_parallel_z and max_parallel_xy
+            qubits: Qubits on the device, identified by their x, y location.
```

Comment:
I thought you put them into a line at the moment? If so it would make sense to allow line qubits or grid qubits but not both.

Reply:
We work with a 2D grid of atoms.